api/database_cluster.go

```diff
@@ -133,6 +192,50 @@ func (e *EverestServer) DeleteDatabaseCluster(
 
 // GetDatabaseCluster retrieves the specified database cluster on the specified kubernetes cluster.
 func (e *EverestServer) GetDatabaseCluster(ctx echo.Context, namespace, name string) error {
+	user, err := rbac.GetUser(ctx)
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, Error{
+			Message: pointer.ToString("Failed to get user from context" + err.Error()),
+		})
+	}
+	db, err := e.kubeClient.GetDatabaseCluster(ctx.Request().Context(), namespace, name)
+	if err != nil {
+		return err
+	}
+	// Check for backup-storage permissions.
+	for _, sched := range db.Spec.Backup.Schedules {
+		if ok, err := e.canGetBackupStorage(user, namespace, sched.BackupStorageName); err != nil {
+			e.l.Error(errors.Join(err, errors.New("failed to check backup-storage permissions")))
+			return err
+		} else if !ok {
+			return ctx.JSON(http.StatusForbidden, Error{
+				Message: pointer.ToString(errInsufficientPermissions.Error()),
+			})
+		}
+	}
+	// Check if the user has permission for the backup-storages used by PITR (if any)?
+	if bsName := pointer.Get(db.Spec.Backup.PITR.BackupStorageName); bsName != "" {
+		if ok, err := e.canGetBackupStorage(user, namespace, bsName); err != nil {
+			e.l.Error(errors.Join(err, errors.New("failed to check backup-storage permissions")))
+			return err
+		} else if !ok {
+			return ctx.JSON(http.StatusForbidden, Error{
+				Message: pointer.ToString(errInsufficientPermissions.Error()),
+			})
+		}
+	}
+	// Check for monitoring-config permissions.
+	if mcName := pointer.Get(db.Spec.Monitoring).MonitoringConfigName; mcName != "" {
+		if ok, err := e.canGetMonitoringConfig(user, namespace, mcName); err != nil {
+			e.l.Error(errors.Join(err, errors.New("failed to check monitoring-config permissions")))
+			return err
+		} else if !ok {
+			return ctx.JSON(http.StatusForbidden, Error{
+				Message: pointer.ToString(errInsufficientPermissions.Error()),
+			})
+		}
+	}
+	// TODO: duplicate call to Kubernetes, need to figure out how to fix this.
 	return e.proxyKubernetes(ctx, namespace, databaseClusterKind, name)
```

Comment:
Won't this 2nd call return the unfiltered list, thus completely bypassing the RBAC filtering we just did?

Reply:
Okay I just found that this is the GET not the LIST but still could't we just remove this 2nd altogether? What's stopping us from doing that?

Reply:
We don't return any list here, this is a `Get` call. If the RBAC fails, it returns a 403.

So it either returns the whole object or an error.

Reply:
> Okay I just found that this is the GET not the LIST but still could't we just remove this 2nd altogether? What's stopping us from doing that?

TBH, I'm not sure. I see that the proxy does some response modification, and I assume we need to do the same for the K8s response - https://github.com/percona/everest/blob/main/api/proxy.go#L194

I'm yet to figure out what this does, so I left a TODO here to unblock the PR and RC-3 release..

Reply:
BTW, I'm doing some code refactoring in a [separate PR](https://github.com/percona/everest/pull/684), and thought I'll address it there. I see we do the duplicate calls in other places too..

Reply:
I see... I'm also not a big fan of the fact that we had to duplicate what's effectively the same filtering for the List and the Get. It would be great if we could find a way to use the same logic on both.

Reply:
But that can be done on the separate refactoring PR